doc/man3/RSA_padding_add_PKCS1_type_1.pod

```diff
@@ -107,15 +123,16 @@ L<ERR_get_error(3)>.
 
 =head1 WARNING
 
-The RSA_padding_check_PKCS1_type_2() padding check leaks timing
+The result of RSA_padding_check_PKCS1_type_2() is a very sensitive
 information which can potentially be used to mount a Bleichenbacher
 padding oracle attack. This is an inherent weakness in the PKCS #1
-v1.5 padding design. Prefer PKCS1_OAEP padding. Otherwise it can
-be recommended to pass zero-padded B<f>, so that B<fl> equals to
-B<rsa_len>, and if fixed by protocol, B<tlen> being set to the
-expected length. In such case leakage would be minimal, it would
-take attacker's ability to observe memory access pattern with byte
-granilarity as it occurs, post-factum timing analysis won't do.
+v1.5 padding design. Prefer PKCS1_OAEP padding. If that is not
+possible, the result of RSA_padding_check_PKCS1_type_2() should be
+checked in constant time if it matches the expected length of the
+plaintext and additionally some application specific consistency
+checks on the plaintext need to be performed in constant time.
+If the plaintext is rejected it must be kept secret which of the
+checks caused the application to reject the message.
```

Comment:
I think this might need an update. I think the 0-padding and expected tlen should still be mentioned.

Reply:
My intention was to cover the expected tlen thing with
"The result of ... should be checked in constant time if it matches the expected length of the plain text"
because passing the expected length in tlen will not do that, it will only match received length <= tlen.
but we need received length == expected length.
And with "additional application specific consistency checks" I mean to cover the case
when the packet is 0-padded, but it may also cover the version check (plain text start with
0x03 0x03) like we do in TLS so it is 0x03-padded, but it serves the same purpose.

Reply:
That was already discussed in the original Bleichenbacher paper,
but checking just 2 bytes and the known length is just a tiny bit too weak.

Reply:
There is this comment:
```
     * caller is encouraged to pass zero-padded message created with
     * BN_bn2binpad. Trouble is that since we can't read out of |from|'s
     * bounds, it's impossible to have an invariant memory access pattern
     * in case |from| was not zero-padded in advance.
```

We should keep suggestion in the manual

Reply:
But you're right about the tlen thing.

Reply:
Okay, I misread that to mean the zero-padded plaintext.
With RSA_private_decrypt(..., NO_PADDING); one would have to remove the zero-padding first.
That would be like shootin one's own knee....

Reply:
I should say, do not remove the zero-padding from RSA_private_decrypt, and that would
even make OAEP vulnerable, right?
It is just so stupid, that my brain refused to see it :-)

Reply:
okay, added a warning that it is not a good idea to remove the zero padding.